### PR TITLE
Removed unnecessary whitespace before text tag

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #50 
@NTaylorMullen @dougbu 

There are a few tests that needs to be updated. Will update them once the product code is signed off.

### CODE REVIEW

File: src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs [hunk 1]
```diff
@@ -411,16 +411,23 @@ private void Statement(Block block)
...
+
                 // MARKUP owns whitespace EXCEPT in DesignTimeMode.
                 PutCurrentBack();
-                PutBack(lastWs);
+
+                // Don't putback the whitespace if it precedes a '<text>' tag.
+                if (!nextSymbol.Content.Equals(SyntaxConstants.TextTagName))
+                {
+                    PutBack(lastWhitespace);
+                }
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs]: looks like this should mostly fix the problem.  but when `nextSymbol.Content.Equals(SyntaxConstants.TextTagName) && lastWhitespace != null` here (i.e. a non-existent `else if`), should we call `Accept(lastWhitespace)`?  right now nothing owns the whitespace before `<text>`

File: test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Source/SimpleCodeBlock.cshtml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿@{
```
Comment (dougbu) [inline test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Source/SimpleCodeBlock.cshtml]: rename this to something more specific e.g. `CodeBlockWithTextElement`.  some name that distinguishes this CodeBlock file in terms of what it contains, not the complexity it lacks.

File: src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs [hunk 3]
```diff
@@ -388,14 +388,14 @@ private void Statement(Block block)
             Span.EditHandler.AcceptedCharacters = AcceptedCharacters.Any;
 
             // Accept whitespace but always keep the last whitespace node so we can put it back if necessary
-            var lastWs = AcceptWhiteSpaceInLines();
-            Debug.Assert(lastWs == null || (lastWs.Start.AbsoluteIndex + lastWs.Content.Length == CurrentLocation.AbsoluteIndex));
+            var lastWhitespace = AcceptWhiteSpaceInLines();
+            Debug.Assert(lastWhitespace == null || (lastWhitespace.Start.AbsoluteIndex + lastWhitespace.Content.Length == CurrentLocation.AbsoluteIndex));
```
Comment (dougbu) [inline src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs]: thanks for the much-improved variable name

**Comment by dougbu** [comment]
:watch:

File: src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs]: Two concerns:

1. This method has some wonkyness to it, for instance the two comments on line 423 and 413 completely contradict each other. Moreover, I don't see why Markup should ever own whitespace even in design time mode. That would end up breaking intellisense for C#. We may want to clarify this with @anurse for what his intentions where when writing this bit.
1. This comment could be completely null and void based on the answer to 1, but this seems like the wrong place to put logic handling the `<text>` tag. There's a few locations where we special case `<text>` in `HtmlMarkupParser.Block` but neither should be impacted by what happens in the C# statement block from a conceptual level. Therefore, IF we can allow C# to own whitespace always in statements that would remove the need for this check entirely.

File: test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Source/SimpleCodeBlock.cshtml [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+    var a = 1; <text>@a</text>
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Source/SimpleCodeBlock.cshtml]: Add a version with a tab

**Comment by NTaylorMullen** [comment]
:watch:

File: src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs (hunk 1 above)
Comment (anurse) [inline src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs]: I think you're right in 1 that there isn't really a need for whitespace before the `<text>` tag to be owned by markup. It probably had something to do with IntelliSense triggering (most of these weird whitespace things did :)) but I can't remember the details now. Double-check that IntelliSense works well around `<text>` tags and I'm happy with the change.

**Comment by ajaybhargavb** [comment]
Updated. @dougbu @NTaylorMullen

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**